src/app/plan/duck/operations.ts

```diff
@@ -242,14 +242,23 @@ const putPlan = planValues => {
       dispatch(Creators.updatePlan(latestPlan));
       const updatedMigPlan = updateMigPlanFromValues(latestPlan, planValues);
 
-      const putRes = await client.put(
-        new MigResource(MigResourceKind.MigPlan, migMeta.namespace),
-        latestPlan.metadata.name,
-        updatedMigPlan
-      );
-      // TODO: Need some kind of retry logic here in case the resourceVersion
-      // gets ticked up in between us getting and putting the mutated object back
-      dispatch(Creators.updatePlan(putRes.data));
+      const fetch_retry: any = async n => {
+        for (let i = 0; i < n; i++) {
+          try {
+            return await client.put(
+              new MigResource(MigResourceKind.MigPlan, migMeta.namespace),
+              latestPlan.metadata.name,
+              updatedMigPlan
+            );
+          } catch (err) {
+            const isLastAttempt = i + 1 === n;
+            if (isLastAttempt) {
+              throw err;
+            }
+          }
+        }
+      };
+      fetch_retry(3).then(res => dispatch(Creators.updatePlan(res.data)));
```

Comment:
This promise without a `catch` actually results in an unhandled exception despite the code being inside of a try catch. I think you might want to turn this into a regular `await`:
```
const res = await fetchRetry(3);
dispatch(Creators.updatePlan(res.data));
```

If that thing throws, it should fall into the `catch` below as you'd expect.

Reply:
very nice - will change it